layers.go

```diff
@@ -382,6 +424,17 @@ func (r *layerStore) lookup(id string) (*Layer, bool) {
 	return nil, false
 }
 
+func (r *layerStore) Size(name string) (int64, error) {
+	layer, ok := r.lookup(name)
+	if !ok {
+		return -1, ErrLayerUnknown
+	}
+	if layer.UncompressedDigest != "" {
+		return layer.UncompressedSize, nil
```

Comment:
Personal-style nitpick, this relationship between `UncompressedDigest != ""` <=> `UncompressedSize` is undocumented.